beginner_bioinf_tools.py

```diff
@@ -0,0 +1,112 @@
+        result = reverse_complement(seqs)
+    else:
+        raise ValueError('Invalid input: unknown operation! Check the last argument.')
+    if len(result) == 1:
+        result = ''.join(result)
+    return result
+
+
+def run_protein_tools(inputs: tuple, options: str = None) -> list or dict:
+    """
+    Produce a list or dictionary according to the option specified
+    arguments:
+    - inputs (tuple): a tuple of inputs
+    - options (str): option name
+    """
+    from modules.protein_tools import calculate_similarity, count_length, info_amino_acid_percentage, find_pattern, convert_to_gene, recode_3letter_to_1letter, check_protein, string_check, check_input
+    check_input(inputs, options)
+    operations = {
+        'similarity': calculate_similarity,
+        'length': count_length,
+        'percentage': info_amino_acid_percentage,
+        'pattern': find_pattern,
+        '3letter_name': recode_3letter_to_1letter,
+        'dna_code': convert_to_gene
+    }
+
+    if options == 'similarity':
+        result = operations[options](inputs[:-2], inputs[-2], inputs[-1])
+        return (result)
+    elif options == 'pattern':
+        result = operations[options](inputs[1:len(inputs)], inputs[0])
+        return (result)
+    elif options == '3letter_name':
+        result = operations[options](inputs[:-1], inputs[-1])
+        return (result)
+    elif options == 'length' or options == 'percentage' or options == 'dna_code':
+        result = []
+        for inpt in inputs:
+            res = operations[options](inpt)
+            result.append(res)
+        return (result)
+    else:
+        raise ValueError('Incorrect options input, please try again')
+
+
+def run_fastq_filtration(seqs: dict, gc_bounds: tuple = (0, 100), length_bounds: tuple = (0, 2**32), quality_threshold: int = 0):
+    """
+    Filtrate a fastq dictionary according to parameters specifies
+    arguments:
+    - seqs (dict): fastq dictionary
+    - gc_bounds (tuple): range of GC content
+    - length_bounds (tuple): range of sequence length
+    - quality_threshold (int): lower limit of average in-sequence quality
+    return:
+    - seqs_filtered (dict): filtered fastq dictionary
+    """
+    from modules.fastq_filtration_tools import check_fastq, check_gc, check_length, check_quality
+    check_fastq(seqs)
+    seqs_filtered = {}
+    if type(gc_bounds) == int:
+        gc_bounds = tuple([0, gc_bounds])
+    if type(length_bounds) == int:
+        length_bounds = tuple([0, length_bounds])
+    for seq_name in seqs:
+        gc_check = False
+        length_check = False
+        quality_check = False
+        gc_check = check_gc(seqs[seq_name][0], gc_bounds)
+        length_check = check_length(seqs[seq_name][0], length_bounds)
+        quality_check = check_quality(seqs[seq_name][1], quality_threshold)
+        if gc_check and length_check and quality_check:
+            seqs_filtered[seq_name] = seqs[seq_name]
+    return seqs_filtered
```

Comment:
1) Тут в цикле можно было бы интерироваться сразу по всем элементам чтобы их использовать.
```suggestion
    for name, (seq, qual) in seqs.items():
        gc_check = False
        length_check = False
        quality_check = False
        gc_check = check_gc(seq, gc_bounds)
        length_check = check_length(seq, length_bounds)
        quality_check = check_quality(qual, quality_threshold)
        if gc_check and length_check and quality_check:
            seqs_filtered[name] = (seq, qual)
    return seqs_filtered
```

2) В целом не нужно пре-создавать 3 переменные заданные в False. Просто записываем в них результаты проверок